dipy/workflows/reconst.py

```diff
@@ -1902,3 +1963,862 @@ def run(
+        out_gfa : string, optional
+            Name of the generalized FA volume to be saved.
+        out_sphere : string, optional
+            Sphere vertices name to be saved.
+        out_b : string, optional
+            Name of the B Matrix to be saved.
+        out_qa : string, optional
+            Name of the Quantitative Anisotropy to be saved.
+
+        References
+        ----------
+        .. footbibliography::
+        """
+        io_it = self.get_io_iterator()
+
+        for (
+            dwi,
+            bval,
+            bvec,
+            maskfile,
+            opam,
+            oshm,
+            opeaks_dir,
+            opeaks_values,
+            opeaks_indices,
+            ogfa,
+            osphere,
+            ob,
+            oqa,
+        ) in io_it:
+            logging.info(f"Loading {dwi}")
+            data, affine = load_nifti(dwi)
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+
+            # If all b-values are smaller or equal to the b0 threshold, it is
+            # assumed that no thresholding is requested
+            if any(mask_non_weighted_bvals(bvals, b0_threshold)):
+                if b0_threshold < bvals.min():
+                    warn(
+                        f"b0_threshold (value: {b0_threshold}) is too low, "
+                        "increase your b0_threshold. It should be higher than the "
+                        f"first b0 value ({bvals.min()}).",
+                        stacklevel=2,
+                    )
+            gtab = gradient_table(
+                bvals, bvecs=bvecs, b0_threshold=b0_threshold, atol=bvecs_tol
+            )
+            mask_vol = load_nifti_data(maskfile).astype(bool)
+
+            n_params = ((sh_order_max + 1) * (sh_order_max + 2)) / 2
+            if data.shape[-1] < n_params:
+                raise ValueError(
+                    f"You need at least {n_params} unique DWI volumes to "
+                    f"compute fiber odfs. You currently have: {data.shape[-1]}"
+                    " DWI volumes."
+                )
+
+            peaks_sphere = (
+                default_sphere if sphere_name is None else get_sphere(name=sphere_name)
+            )
+
+            logging.info("FORECAST computation started.")
+            forecast_model = ForecastModel(
+                gtab,
+                sh_order_max=sh_order_max,
+                lambda_lb=lambda_lb,
+                dec_alg=dec_alg,
+                sphere=peaks_sphere.vertices,
+                lambda_csd=lambda_csd,
+            )
+
+            peaks_forecast = peaks_from_model(
+                model=forecast_model,
+                data=data,
+                sphere=peaks_sphere,
+                relative_peak_threshold=relative_peak_threshold,
+                min_separation_angle=min_separation_angle,
+                mask=mask_vol,
+                return_sh=True,
+                sh_order_max=sh_order_max,
+                normalize_peaks=True,
```

Comment:
`normalize_peaks` Should be provided to the user as we can control it with a variable.

Reply:
same comment as above:

For now, I do not think it is necessary to provide access to this variable.

Later on, if requested, we could.